### PR TITLE
feat: Onboard San Francisco Bikeshare Trips

### PR DESCRIPTION
**Pull request by nlarge-google**
## Checklist

Note: Delete items below that aren't applicable to your pull request.

- [x] Please merge this PR for me once it is approved.
- [x] If this PR adds or edits a dataset or pipeline, it was reviewed and approved by the Google Cloud Public Datasets team beforehand.
- [x] If this PR adds or edits a dataset or pipeline, I put all my code inside  `san_francisco_bikeshare_trips/bikeshare_trips` and nothing outside of that directory.
- [x] This PR is appropriately labeled.

### CODE REVIEW

**Comment by nlarge-google** [comment]
Here is the "complex" pipeline that you needed me to work on. Was a lot of fun! Please review. Thanks!

**Comment by adlersantos** [comment]
@nlarge-google Have you done the renaming from `san_francisco_bikeshare_trips` to `san_francisco_bikeshare` for the dataset name?

**Review by adlersantos — CHANGES_REQUESTED** [review]

File: datasets/san_francisco_bikeshare_trips/_images/run_csv_transform_kub/csv_transform.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+        "start_station_terminal": np.int_,
+        "end_date": np.str_,
+        "end_station_name": np.str_,
+        "end_station_terminal": np.str_,
+        "bike_number": np.int_,
+        "subscription_type": np.str_,
+        "zip_code": np.str_,
+    }
+
+    tripdata_names = [
```
[the remark this answers is not in the document]
Comment (nlarge-google) [reply datasets/san_francisco_bikeshare_trips/_images/run_csv_transform_kub/csv_transform.py:76]: @adlersantos I answered this question during our code review meeting but for documentative purposes I will answer it formally here.  _tripdata_names_ refers to the calendar years 2014-2016 format of the source file.  These source files changed considerably in 2017 and so their dataframe content is implemented in the _trip_data_names_ dataframe.  The original ETL process reflects this architectural approach and as such has been replicated during this migration.

**Comment by adlersantos** [comment]
> @nlarge-google Have you done the renaming from `san_francisco_bikeshare_trips` to `san_francisco_bikeshare` for the dataset name?

FYI

File: datasets/san_francisco_bikeshare_trips/bikeshare_trips/pipeline.yaml [hunk 2]
```diff
@@ -80,7 +80,8 @@ dag:
         bucket: "{{ var.value.composer_bucket }}"
         source_objects: ["data/san_francisco_bikeshare_trips/bikeshare_trips/data_output.csv"]
         source_format: "CSV"
-        destination_project_dataset_table: "san_francisco_bikeshare_trips.bikeshare_trips"
+        destination_project_dataset_table: "{{ var.json.san_francisco_bikeshare_trips.container_registry.bikeshare_trips_destination_table }}"
```
Comment (adlersantos) [inline datasets/san_francisco_bikeshare_trips/bikeshare_trips/pipeline.yaml:83]: Please change to `san_francisco_bikeshare` as well

File: datasets/san_francisco_bikeshare_trips/bikeshare_trips/bikeshare_trips_dag.py [hunk 3]
```diff
@@ -76,7 +76,7 @@
             "data/san_francisco_bikeshare_trips/bikeshare_trips/data_output.csv"
         ],
         source_format="CSV",
-        destination_project_dataset_table="san_francisco_bikeshare_trips.bikeshare_trips",
+        destination_project_dataset_table="{{ var.json.san_francisco_bikeshare_trips.container_registry.bikeshare_trips_destination_table }}",
```
Comment (adlersantos) [inline datasets/san_francisco_bikeshare_trips/bikeshare_trips/bikeshare_trips_dag.py:79]: Please change to san_francisco_bikeshare as well

**Comment by nlarge-google** [comment]
@adlersantos I have taken the code from this PR and put it into #240 so I think it may be best to close or cancel this PR.  Please let me know your thoughts.  Thanks.

File: datasets/san_francisco_bikeshare_trips/bikeshare_trips/pipeline.yaml (hunk 2 above)
Comment (nlarge-google) [reply datasets/san_francisco_bikeshare_trips/bikeshare_trips/pipeline.yaml:83]: @adlersantos Please see my last comment

File: datasets/san_francisco_bikeshare_trips/bikeshare_trips/bikeshare_trips_dag.py (hunk 3 above)
Comment (nlarge-google) [reply datasets/san_francisco_bikeshare_trips/bikeshare_trips/bikeshare_trips_dag.py:79]: @adlersantos Please see my last comment

**Comment by adlersantos** [comment]
@

> @adlersantos I have taken the code from this PR and put it into #240 so I think it may be best to close or cancel this PR. Please let me know your thoughts. Thanks.

Sure I'll close this PR. Thanks!

**Pull request closed without merging**